src/ProcessManager.php

```diff
@@ -745,38 +751,41 @@ protected function checkChangedFiles($restartSlaves = true)
                 if ($this->filesLastMTime[$filePath] !== $currentFileMTime) {
                     $this->filesLastMTime[$filePath] = $currentFileMTime;
 
-                    $md5 = md5_file($filePath);
+                    $md5 = md5_file($filePath, true);
                     if (!isset($this->filesLastMd5[$filePath]) || $md5 !== $this->filesLastMd5[$filePath]) {
                         $this->filesLastMd5[$filePath] = $md5;
-                        $reload = true;
-
-                        //since chances are high that this file will change again we
-                        //move this file to the beginning of the array, so next check is way faster.
-                        unset($this->filesToTrack[$idx]);
```

Comment:
why remove the array reordering?

Reply:
reordering doesn't make sense, since we do a full scan of the tracked files

Reply:
I don't think so- we're breaking the loop on the first change found?

Reply:
> we're breaking the loop on the first change found?

No, we are not. I removed the break statement (see below).  Otherwise, we can't update the modification time of all tracked files.